tests/kernel/futex/src/main.c

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <irq_offload.h>
+
+/* Macro declarations */
+#define TOTAL_THREADS_WAITING (5)
+#define STACK_SIZE (1024 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+#define futex_wake_from_isr(futex) irq_offload(futex_isr_wake, futex)
+
+/******************************************************************************/
+/* declaration */
+K_THREAD_STACK_DEFINE(stack_1, STACK_SIZE);
+K_THREAD_STACK_ARRAY_DEFINE(multiple_stack, TOTAL_THREADS_WAITING, STACK_SIZE);
+
+struct k_mem_domain futex_domain;
+K_APPMEM_PARTITION_DEFINE(futex_partition);
+K_APP_BMEM(futex_partition) int timeout;
```

Comment:
strictly speaking, you could just add these globals to the ztest partition using ZTEST_BMEM.
it might make this test easier to run on systems that only have 8 MPU regions.